docs/usage/self-hosted-configuration.md

```diff
@@ -912,6 +912,34 @@ It also may mean that ignored directories like `node_modules` can be preserved a
 
 ## platform
 
+## platformOptions
```

Comment:
I'm confused by why this is in both self-hosted and repo docs. Why is that?

Reply:
While `platformOptions` can appear in both configs, the specific child options within it are constrained depending on whether they are part of the admin (self-hosted) or repo config.

For example:

``` platformOptions: { gitlabIgnoreApprovals: true } ``` can be used in both the admin (self-hosted) and repository configurations.
But, ``` platformOptions: { platformVersion: '1.2.3' } ``` can only be used in the admin (self-hosted) configuration because `platformVersion` is specific to admin config.

To make this distinction clearer, I've added documentation for `platformOptions` in both the self-hosted and repository docs. In each, I've listed which children are allowed in the respective configs (admin or repo).

Reply:
I don't like this approach - too much chance of confusion or mistake.

Reply:
@rarkins Any other suggestion how to handle platform options? Maybe naming the other `repoOptions` for repo configs?

Reply:
What was the idea behind creating this combined platformOptions option? to which functions are they passed? e.g. for initPlatform() only the global ones are passed, but for ensurePr() both global and repo are passed?

Reply:
The idea was to organise existing platform options under one parent to make it easier for users to find them in the docs.
Global ones are passed to `initPlatform` and repo ones are passed to `ensurePr`. 

The global ones are filtered out in the `getPlatformPrOptions` fn before being passed to `ensurePr`.

I agree, though that this is confusing.

Reply:
Could it just be `prOptions` for short? `platformOptions` are passed to `initPlatform()` and `prOptions` are passed to `ensurePr()`

Reply:
sounds good 👍